library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -577,16 +586,28 @@ public synchronized void onSetUserCompleted(ClientConnectionCallback callback) {
 
     private synchronized void connect() {
         Log.i(TAG, "client.connect was called");
-        tokenProvider.getToken(userToken -> {
+
+        if (!anonymousConnection) {
```

Comment:
please reverse logic, no need to make code harder to read :)

```java
if (anonymousConnection) {
    // do anon stuff
} else {
    // do not anon stuff
}
```

Reply:
corrected